crates/tabby-index/src/code/index.rs

```diff
@@ -165,6 +172,48 @@ fn is_valid_file(file: &SourceCode) -> bool {
         && file.number_fraction <= MAX_NUMBER_FRACTION
 }
 
+fn create_document_with_commit(doc: &TantivyDocument, commit: &str) -> TantivyDocument {
```

Comment:
can this method being somewhat refactored into `TantivyDocBuilder`? e.g `TantivyDocBuilder::backfill_doc_attribute`